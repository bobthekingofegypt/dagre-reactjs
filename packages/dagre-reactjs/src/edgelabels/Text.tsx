import * as React from 'react';

import { CustomEdgeLabelProps } from '../types';

export const Text: React.FC<CustomEdgeLabelProps> = ({ edgeMeta }) => {
  return (
    <text
      className={edgeMeta.styles.label.className}
      style={edgeMeta.styles.label.styles || {}}
    >
      <tspan xmlSpace="preserve" dy="1em" x="1">
        {edgeMeta.label || ''}
      </tspan>
    </text>
  );
};
