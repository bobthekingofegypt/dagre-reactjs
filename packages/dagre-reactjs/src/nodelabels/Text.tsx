import * as React from 'react';

import { CustomNodeLabelProps } from '../types';

export const Text: React.FC<CustomNodeLabelProps> = ({ node }) => {
  return (
    <text
      className={node.styles.label.className}
      style={node.styles.label.styles || {}}
    >
      <tspan xmlSpace="preserve" dy="1em" x="1">
        {node.label}
      </tspan>
    </text>
  );
};
