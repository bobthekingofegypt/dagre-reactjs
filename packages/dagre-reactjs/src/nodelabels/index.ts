import { NodeLabelDefinition, NodeLabelsDefinition } from '../types';
import { Text } from './Text';

export const builtInNodeLabels: NodeLabelsDefinition = {
  text: {
    renderer: Text,
    html: false,
  },
};

export const getNodeLabel = (
  type: string,
  nodeLabels: NodeLabelsDefinition
): NodeLabelDefinition => {
  if (nodeLabels[type]) {
    return nodeLabels[type];
  }

  throw Error(`node label ('${type}') not found in builtins or custom`);
};
