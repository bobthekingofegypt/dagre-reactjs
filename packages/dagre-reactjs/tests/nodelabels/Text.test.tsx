import { shallow } from 'enzyme';
import * as React from 'react';

import { Text } from '../../src/nodelabels/Text';
import { getDefaultNodeOptions } from '../stubs';

describe('Text', () => {
  it('renders correctly', () => {
    const node = getDefaultNodeOptions('1', 'test label');
    const wrapper = shallow(<Text node={node} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the given label', () => {
    const node = getDefaultNodeOptions('1', 'test label');
    const wrapper = shallow(<Text node={node} />);
    const tspanElement = wrapper.find('tspan');
    expect(tspanElement.text()).toBe('test label');
  });

  it('renders the empty string label when empty label passed', () => {
    const node = getDefaultNodeOptions('1', '');
    const wrapper = shallow(<Text node={node} />);
    const tspanElement = wrapper.find('tspan');
    expect(tspanElement.text()).toBe('');
  });

  it('uses the custom css classname when passed', () => {
    const node = getDefaultNodeOptions('1', '');
    node.styles.label.className = 'test-classname';
    const wrapper = shallow(<Text node={node} />);

    const text = wrapper.find('text');
    expect(text.hasClass('test-classname')).toBe(true);
  });
});
