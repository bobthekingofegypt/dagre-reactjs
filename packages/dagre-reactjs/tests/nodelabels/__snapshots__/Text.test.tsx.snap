// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<text
  style={Object {}}
>
  <tspan
    dy="1em"
    x="1"
    xmlSpace="preserve"
  >
    test label
  </tspan>
</text>
`;
